File: Platformer/ts/_generated/scripts/Title.ts
import { registerScript } from "julgame/src/engine/runtime/scriptRegistry";
import { registerScriptSounds } from "julgame/src/engine/runtime/scriptRegistry";


    // using JulGame

    export class Title {
        fade
        parent
        textBox

        constructor() {
            

            this.fade = true
            this.parent = null
            this.textBox = null

        }
    }

    function JulGame_initialize_Title(self: Title) {
        //(globalThis as any).JulGame.MainLoopModule.enable_profiling()
        self.textBox = (globalThis as any).MAIN.scene.uiElements[0]
    }

    function JulGame_update_Title(self: Title, deltaTime) {
        try {
            if (self.fade) {
                self.textBox.color = [self.textBox.color[0], self.textBox.color[1], self.textBox.color[2], self.textBox.color[3] - 1]
                self.textBox.text = self.textBox.text
                if (self.textBox.color[3] <= 25) {
                    self.fade = false
                }
            } else {
                self.textBox.color = [self.textBox.color[0], self.textBox.color[1], self.textBox.color[2], self.textBox.color[3] + 1]
                self.textBox.text = self.textBox.text
                if (self.textBox.color[3] >= 250) {
                    self.fade = true
                }
            }

            if ((globalThis as any).JulGame.InputModule.get_button_pressed((globalThis as any).MAIN.input, "RETURN")) {
                (globalThis as any).JulGame.change_scene("level_1.json")
                // sound = (globalThis as any).JulGame.create_sound_source(self.parent, (globalThis as any).JulGame.SoundSourceModule.SoundSource(Number(-1), false, "confirm-ui.wav", 50))
                // (globalThis as any).JulGame.Component.toggle_sound(sound)
            }
        } catch (e) {
            console.error(String(e))


        }
    }
registerScript("Title", {
    create: () => new Title(),
    initialize: JulGame_initialize_Title,
    update: JulGame_update_Title,
});
registerScriptSounds(["confirm-ui.wav"]);

File: Platformer/scripts/Title.jl
module TitleModule
    using JulGame

    mutable struct Title <: Script
        fade
        parent
        textBox

        function Title()
            this = new()

            this.fade = true
            this.parent = C_NULL
            this.textBox = C_NULL

            return this
        end
    end

    function JulGame.initialize(this::Title)
        #JulGame.MainLoopModule.enable_profiling()
        this.textBox = MAIN.scene.uiElements[1]
    end

    function JulGame.update(this::Title, deltaTime)
        try
            if this.fade 
                this.textBox.color = (this.textBox.color[1], this.textBox.color[2], this.textBox.color[3], this.textBox.color[4] - 1)
                this.textBox.text = this.textBox.text
                if this.textBox.color[4] <= 25
                    this.fade = false
                end
            else
                this.textBox.color = (this.textBox.color[1], this.textBox.color[2], this.textBox.color[3], this.textBox.color[4] + 1)
                this.textBox.text = this.textBox.text
                if this.textBox.color[4] >= 250
                    this.fade = true
                end
            end

            if JulGame.InputModule.get_button_pressed(MAIN.input, "RETURN")
                JulGame.change_scene("level_1.json")
                # sound = JulGame.create_sound_source(this.parent, JulGame.SoundSourceModule.SoundSource(Int32(-1), false, "confirm-ui.wav", Int32(50)))
                # JulGame.Component.toggle_sound(sound)
            end
        catch e
            @error string(e)
            Base.show_backtrace(stdout, catch_backtrace())
            rethrow(e)
        end
    end
end # module